### PR TITLE
GAN

### PR DESCRIPTION
**Pull request by vitskvara**

### CODE REVIEW

**Comment by vitskvara** [comment]
I implemented the WGAN model as well, but without run scripts, as it does not seem to be applicable to AD.

File: src/models/gan.jl [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+using Distributions
+using DistributionsAD
+using StatsBase
+using Random
+
+"""
+	gan_constructor(;idim::Int=1, zdim::Int=1, activation = "relu", hdim=128, nlayers::Int=3, 
+		init_seed=nothing, prior="normal", pseudoinput_mean=nothing, k=1, kwargs...)
+
+Constructs a classical variational autoencoder.
```
Comment (janfrancu) [inline src/models/gan.jl]: Incorrect docs.

File: scripts/count_runs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+function list_files(datapath)
+	fs = readdir(datapath)
+	fs = filter(f->!(occursin("model_", f)), fs)
+	fs = filter(f->!(occursin("#", f)), fs)
+	return unique(fs)
+end
+function _check_validity(file)
+	data = load(file)
+	return (length(data[:val_labels]) == length(data[:val_scores])) && length(data[:tst_labels]) == length(data[:tst_scores])
```
Comment (janfrancu) [inline scripts/count_runs.jl]: Checking `any(isnan.(___))` may be also useful.

Comment (vitskvara) [reply scripts/count_runs.jl]: True, will add that.

File: src/models/gan.jl (hunk 1 above)
Comment (vitskvara) [reply src/models/gan.jl]: Thanks.

**Pull request merged**